src/reanimated2/component/KeyboardAvoidingView.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { forwardRef, useCallback } from 'react';
+import type {
+  KeyboardAvoidingViewProps,
+  LayoutRectangle,
+  View,
+  ViewProps,
+} from 'react-native';
+import { StyleSheet, useWindowDimensions } from 'react-native';
+import { AnimatedView as ReanimatedView } from './View';
+import {
+  useAnimatedKeyboard,
+  useAnimatedStyle,
+  runOnUI,
+  useSharedValue,
+} from '../';
+
+let useHeaderHeight: () => number;
+try {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  useHeaderHeight = require('@react-navigation/elements').useHeaderHeight;
+} catch (e) {
+  useHeaderHeight = () => 0;
+}
+
+type Props = ViewProps & KeyboardAvoidingViewProps;
```

Comment:
`KeyboardAvoidingViewProps` type extends `ViewProps` out of the box from what I can see. There is no need for this type.
```suggestion
```